src/backends/fvm_gpu.hpp

```diff
@@ -240,6 +374,46 @@ void assemble_matrix(matrix_update_param_pack<T, I> params, T dt) {
     }
 }
 
+template <typename T, typename I, typename Stack>
+__global__
+void test_thresholds(
+    float t, float t_prev, int size,
+    Stack& stack,
+    I* is_spiking, T* prev_values,
+    const I* index, const T* values, const T* thresholds)
+{
+    int i = threadIdx.x + blockIdx.x*blockDim.x;
+
+    bool crossed = false;
+    float crossing_time;
+
+    if (i<size) {
+        // Test for threshold crossing
+        const auto v_prev = prev_values[i];
+        const auto v      = values[index[i]];
+        const auto thresh = thresholds[i];
+
+        if (!is_spiking[i]) {
+            if (v>=thresh) {
+                // The threshold has been passed, so estimate the time using
+                // linear interpolation
+                auto pos = (thresh - v_prev)/(v - v_prev);
+                crossing_time = t_prev + pos*(t - t_prev);
+
+                is_spiking[i] = 1;
+                crossed = true;
+            }
+        }
+        else if (v<thresh) {
+            is_spiking[i]=0;
+        }
+
+        prev_values[i] = v;
+    }
+
+    stack.push_back({I(i), crossing_time}, crossed);
```

Comment:
I can see the argument either way, though if we keep the conditional push, can we call it `push_conditional`? My personal preference is to keep it simple and then change, but it's only a preference.